MIGRATING.md

```diff
@@ -0,0 +1,59 @@
+# Okta Java Authentication SDK Migration Guide
+ 
+This SDK uses semantic versioning and follows Okta's [library version policy](https://developer.okta.com/code/library-versions/). In short, we do not make breaking changes unless the major version changes!
+
+## Migrating from 1.x.x to 2.0.0
+
+Version 2.0.0 of this SDK introduces few breaking changes from previous versions. 
+In addition to few new classes/interfaces, some existing classes/interfaces are no longer backward compatible.
+
+### Package `com.okta.authn.sdk.resource`
+
+- Replaced `com.okta.sdk.resource.user.factor.FactorProfile` interface with `com.okta.authn.sdk.resource.FactorProfile` interface.
+- Replaced `com.okta.sdk.resource.user.factor.FactorProvider` interface with `com.okta.authn.sdk.resource.FactorProvider` interface.
+- Replaced `com.okta.sdk.resource.user.factor.FactorType` interface with `com.okta.authn.sdk.resource.FactorType` interface.
+
+Note: Old interfaces above were pulled in from **okta-sdk-java** Management SDK hitherto. 
+These are now migrated to reside locally within this Authentication SDK. 
+
+- Added below extensions of `com.okta.authn.sdk.resource.FactorProfile` interface:
+  ```java
+  - CallFactorProfile
+  - EmailFactorProfile
+  - HardwareFactorProfile
+  - PushFactorProfile
+  - SecurityQuestionFactorProfile
+  - SmsFactorProfile
+  - TokenFactorProfile
+  - TotpFactorProfile
+  - U2fFactorProfile
+  - WebAuthnFactorProfile
+  - WebFactorProfile
+  ```
+ 
+Below SDK classes/interfaces are **deprecated** and will be removed from this project.
+
+```
+- com.okta.sdk.http.UserAgentProvider
+- com.okta.authn.sdk.impl.http.AuthnSdkUserAgentProvider
+- com.okta.sdk.cache.CacheManager
+- com.okta.sdk.client.AuthenticationScheme
+- com.okta.sdk.impl.http.authc.RequestAuthenticatorFactory
```

Comment:
This project didn't use cache or an AuthenticationScheme.
The other classes are implementation details.

IMHO, this whole "deprecated section could be removed

Reply:
makes sense, will remove this whole section.

Reply:
sure, will remove.